proto/cosmos/crypto/keyring/types.proto

```diff
@@ -0,0 +1,36 @@
+syntax = "proto3";
+package cosmos.crypto.keyring.v1;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/any.proto";
+import "cosmos/crypto/hd/hd.proto";
+
+option go_package = "github.com/cosmos/cosmos-sdk/crypto/keyring";
+option (gogoproto.goproto_getters_all) = false;
+
+// Record is used for keyring migration from legacyInfo to proto.
+message Record {
+  string name = 1;
+  google.protobuf.Any pub_key = 2;
+
+  // Record contains one of the following items
+  oneof item {
+    Local local = 3;
+    Ledger ledger  = 4;
+    Empty empty  = 5;
```

Comment:
Why are we using empty? I think using `Multi` and `Offline` would be clearer, even both of them are empty structs